lib/resourcedelete/apiext.go

```diff
@@ -10,25 +10,23 @@ import (
 )
 
 // DeleteCustomResourceDefinitionv1 checks the given resource for a valid delete annotation. If found
-// and in UpdatingMode it will issue a delete request or provide status of a previousily issued delete request.
-// If not in UpdatingMode it simply returns an indication that the delete annotation was found. An error is
-// returned if an invalid annotation is found or an error occurs during delete processing.
+// it checks the status of a previousily issued delete request. If delete has not been
+// requested and in UpdatingMode it will issue a delete request.
 func DeleteCustomResourceDefinitionv1(ctx context.Context, client apiextclientv1.CustomResourceDefinitionsGetter,
 	required *apiextv1.CustomResourceDefinition, updateMode bool) (bool, error) {
 
 	if delAnnoFound, err := ValidDeleteAnnotation(required.Annotations); !delAnnoFound || err != nil {
 		return delAnnoFound, err
-	} else if !updateMode {
-		return true, nil
 	}
+	existing, err := client.CustomResourceDefinitions().Get(ctx, required.Name, metav1.GetOptions{})
```

Comment:
There's no real reason to move this higher, right?

Reply:
Not required but I did it so this matches `resourcedelete` delete methods. Also, a minor thing, but now `resource` is declared closer to where it's used.